Library/Util.ts

```diff
@@ -151,5 +154,29 @@ class Util {
 
         return map;
     }
+
+    /**
+     * Checks if a request url is not on a excluded domain list 
+     * and if it is safe to add correlation headers (x-ms-request-source-ikey, x-ms-request-target-ikey)
+     */
+    public static canIncludeCorrelationHeader(client: Client, options: http.RequestOptions | http.ServerResponse) {
+        let excludedDomains = client && client.config && client.config.correlationHeaderExcludedDomains;
+        if (!excludedDomains || excludedDomains.length == 0 || !options) {
+            return true;
+        }
+
+        let url = (<http.RequestOptions>options).host ||
+                ((<http.ServerResponse>options).getHeader && (<http.ServerResponse>options).getHeader("host"));
+
+        for (let i = 0; i < excludedDomains.length; i++) {
+            let matches = url ? url.match(excludedDomains[i]) : null;
```

Comment:
I'm concerned that it's very easy to add a string to the `config.correlationHeaderExcludedDomains` array. (Unit tests actually do that.) Then a string gets interpreted directly as a regular expression, which is not necessarily what would be expected, for example a period in the string will match *any* character.

I'd suggest either (or a combination) of the following approaches:
 1. Check if the exclusion item is an instance of a `RegExp`, if not just do a plain string `indexOf` test. 
 2. Use glob matching instead of regular expressions, e.g. `"*.blob.core.windows.net"`. To avoid adding a dependency to a separate glob package, you can implement very simple glob syntax like: `new RegExp(str.replace(/\*/g, '.*')`